saleor/cart/views.py

```diff
@@ -51,14 +50,8 @@ def index(request, cart):
 @decorators.get_or_create_db_cart()
 def add_to_cart(request, cart, product_id):
     product = get_object_or_404(Product, pk=product_id)
-    form_class = get_form_class_for_product(product)
-
-    form = form_class(
-        data=request.POST or None, product=product, cart=cart,
-        discounts=request.discounts)
-    if form.is_valid():
-        form.save()
-    else:
+    form = process_purchase(request, cart, product)
```

Comment:
When is this view used? What are the user expectations? It seems to have the same problems as the previous version.